plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1769,6 +1775,385 @@ public void testSupportedNonPartitionedColumnMappingNoneWrites()
         testSupportedNonPartitionedColumnMappingWrites("write_stats_as_json_column_mapping_none", false);
     }
 
+    @Test
+    public void testCreateOrReplaceTableOnNonExistingTable()
+    {
+        String tableName = "create_or_replace_table" + randomNameSuffix();
+        try {
+            assertUpdate("CREATE OR REPLACE TABLE " + tableName + " (id BIGINT)");
+            assertLatestTableOperation(tableName, CREATE_OR_REPLACE_TABLE_OPERATION);
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableAsSelectOnNonExistingTable()
+    {
+        String tableName = "create_or_replace_table_as_select_" + randomNameSuffix();
+        try {
+            assertUpdate("CREATE OR REPLACE TABLE " + tableName + " AS SELECT 1 as colA", 1);
+            assertLatestTableOperation(tableName, CREATE_OR_REPLACE_TABLE_AS_OPERATION);
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Test
+    public void testCreateOrReplaceTableAsSelectWithSwappedColumns()
+    {
+        testCreateOrReplaceTableAsSelectWithSwappedColumns(ColumnMappingMode.ID);
+        testCreateOrReplaceTableAsSelectWithSwappedColumns(ColumnMappingMode.NAME);
+        testCreateOrReplaceTableAsSelectWithSwappedColumns(ColumnMappingMode.NONE);
+    }
+
+    private void testCreateOrReplaceTableAsSelectWithSwappedColumns(ColumnMappingMode columnMappingMode)
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_create_or_replace_with_column",
+                " AS SELECT 'abc' colA, BIGINT '1' colB")) {
```

Comment:
nit: Leading a space in the 3rd argument is redundant as TestTable adds it internally.